content/blog/2018-12-25-darktable-2.6/2018-12-25-darktable-2.6.md

```diff
@@ -0,0 +1,335 @@
+
+The interface has a lot more buttons than "spot removal", but
+everything you could do with the later is also possible in a similar
+way with retouch.
+
+  <div style="text-align: center">
+![Retouch module overview]({filename}rt-overview.png)
+  </div>
+
+Just like with "spot removal", you may select a shape (circle,
+ellipse, path, or brush, which wasn't available with "liquify") and
+click on a part of the image you want to hide. The module will clone
+another part of the image to hide it. Drag instead of clicking to
+choose the source of the clone, or adjust the controls after the fact.
+
+Many details will make your life easier than before:
+
+#### Seamless cloning, better results with less efforts
+
+By default, the clone uses a "seamless cloning" algorithm (borrowed
+from GIMP's "heal" tool), which adapts the source to the context
+where it is cloned. You don't need to clone *exactly* the right
+piece of image. Let's take a common example: a small defect in a
+not-completely-uniform sky:
+
+  <div style="text-align: center">
+![Retouch example]({filename}rt-original.png)
+  </div>
+
+A bad attempt to fix this with the clone tool would give:
+
+  <div style="text-align: center">
+![Retouch clone (with controls)]({filename}rt-clone.png)
+  </div>
+
+The piece of sky copied to mask the defect is a bit darker than the
+place where it's copied. It's not obvious while the controls are
+displayed on the image, but the final image is really bad:
+
+  <div style="text-align: center">
+![Retouch clone (without controls)]({filename}rt-clone-nocontrol.png)
+  </div>
+
+The same retouch with the new "heal" tool
+(![heal tool]({filename}rt-heal-icon.png)) gives this:
+
+  <div style="text-align: center">
+![Retouch heal (with controls)]({filename}rt-heal.png)
+  </div>
+
+This time, the final image is undistinguishable from a real sky
+without defect:
+
+  <div style="text-align: center">
+![Retouch heal (without controls)]({filename}rt-heal-nocontrol.png)
+  </div>
+
+Even when cloning pieces of images of totally different colors, the
+"heal" tool behaves surprisingly well. Let's push the module a bit:
+
+  <div style="text-align: center">
+![Retouch heal on different colors]({filename}rt-heal-red-white-blue.png)
+  </div>
+
+The white patch is cloned to the blue shirt, the blue shirt to the red
+one, and the red one to the white one. Each time, the local contrast
+is kept, but the overall color and luma of the patch is adapted to fit
+the destination.
+
+The basic "clone" tool (![clone tool]({filename}rt-clone-icon.png)) is
+still available for the rare cases where you would want it.
+
+#### Fill and blur, when you don't have anything to clone
+
+In addition to the "clone" and "heal" tools (which work only when
+you have a piece of image to duplicate on the spot to remove),
+retouch provides "fill" (![fill tool]({filename}rt-fill-icon.png))
+(pick a color, and fill a shape with this
+color) and "blur" (![blur tool]({filename}rt-blur-icon.png))
```

Comment:
The default mode for 'fill' is erase, it will erase the details on a detail scale. Is like painting with middle gray on GIMP & the wavelet decompose. Internally the fill (on erase mode) paints with black, you can see it by painting on the original or residual image, this is because the GIMP's wd add middle gray to the detail scales for better viewing, here in dt that's not needed as we have the levels, so painting with black gives the same result.

Reply:
Thanks, I've added a bit more details (essentially: use erase for details layers and color picker on the residual image).